### ISSUE TITLE
`OptionForceIntegerNumbersEvaluationsAsDoubleByDefault` leads to exception when dividing 

### ISSUE THREAD

**Issue by midgleyc** [codingseb/ExpressionEvaluator#110]
The code

```
var evaluator = new ExpressionEvaluator()
{
    OptionForceIntegerNumbersEvaluationsAsDoubleByDefault = true 
};

string expression = "3/Math.Round(Avg(1,2),MidpointRounding.AwayFromZero)";

Console.WriteLine(expression);
Console.WriteLine(evaluator.Evaluate(expression));
```

gives the error
> Unhandled exception. Microsoft.CSharp.RuntimeBinder.RuntimeBinderException: Operator '/' cannot be applied to operands of type 'double' and 'decimal'

It worked in `1.4.20` and first stopped working in `1.4.21`. `Math.Round` is picking the decimal overload when I'd expect it to pick the double overload. It works without `OptionForceIntegerNumbersEvaluationsAsDoubleByDefault`.

**Comment by codingseb** [comment]
Hello @midgleyc, thanks to reporting this bug.
I just have an other behavior on this one.
I do not have the `Microsoft.CSharp.RuntimeBinder.RuntimeBinderException` but it returns `2` instead.
I tested it with version 1.4.20 and 1.4.30

Anyway it should returns `1.5` so I will investigate why it return `2`

**Comment by codingseb** [comment]
Ok after some investigations I understood that it was a bug in method override selection. 
And depending on which version of .net you use it selected an other method it depends on the order it received the potential methods list. So in your case I think it took `Math.Round(decimal, MidpointRounding)` and in my case `Math.Round(double, int32)` and automatically converted the enum in it's corresponding int value. 
I made a correction and published version 1.4.31.0

So thanks again to finding out this bug
